clang/test/C/C99/n717.c

```diff
@@ -0,0 +1,69 @@
+// RUN: %clang_cc1 -verify -std=c99 %s
+// RUN: %clang_cc1 -verify -std=c99 -fno-dollars-in-identifiers %s
+
+/* WG14 N717: Clang 17
+ * Extended identifiers
+ */
```

Comment:
Do you know what was not supported before?

Reply:
UCNs weren't supported in C89 but were added in C99, so this is testing UCNs as specified in https://www.open-std.org/jtc1/sc22/wg14/www/docs/n717.htm but based on whatever was in the final text of the C99 standard (so it incorporates changes from NB comments, etc).

Reply:
I mean, before clang 17. I don't remember we changed anything there (for C99) in a while.

Reply:
Ah! We were missing some diagnostics: https://godbolt.org/z/hEKsEKqzT